ctr/src/lib.rs

```diff
@@ -82,18 +82,17 @@ impl<C> Ctr128<C>
         C::ParBlocks: ArrayLength<GenericArray<u8, U16>>,
 {
     /// Create new CTR mode instance using initialized block cipher.
-    pub fn from_cipher(cipher: C, nonce: &Nonce<Self>)-> Self {
-        assert!(Self::block_size() <= core::u8::MAX as usize);
-        // see https://github.com/rust-lang/rust/issues/55044
-        let nonce = conv_be(unsafe {
-            ptr::read_unaligned(
-                nonce as *const Nonce<Self> as *const [u64; 2]
-            )
-        });
+    pub fn from_cipher(cipher: C, nonce: &GenericArray<u8, U16>) -> Self {
```

Comment:
Sidebar, but: it seems to me like it'd make a lot more sense to use `nonce: &[u8; 16]` here. I changed it to an explicit `GenericArray` for clarity, but since this is `Ctr128` it will always have a 128-bit/16-byte nonce.

Reply:
This function is used in two places, so it's done to reduce amount of unsafe casts from `GenericArray` to array. Safe casts via `Into` trait were only added in `generic-array v0.12.3`.